### PR TITLE
Fix Form Recognizer Readme Issue 

### PR DESCRIPTION
**Pull request by Menghua1**
Fix [#33776](https://github.com/Azure/azure-sdk-for-java/issues/33776)

@joshfree, @mayurid, @achandmsft and @mssfang for notification.

### CODE REVIEW

**Comment by mssfang** [comment]
I will approve it since most of the other client libraries use the same link for API reference. But I will wait for @samvaity approval.

**Review by samvaity — APPROVED** [review]

**Pull request merged**
